contentcuration/contentcuration/frontend/channelList/views/Channel/components/StudioChannelCard.vue

```diff
@@ -0,0 +1,431 @@
+            </div>
+          </div>
+          <div class="footer-right">
+            <div
+              :ref="`detailIcon${index}`"
+              class="details-link"
+            >
+              <router-link
+                :data-testid="`details-button-${index}`"
+                :to="channelDetailsLink(channel)"
+                @click.native.stop
+              >
+                <KIcon
+                  class="details-icon"
+                  :color="$themeTokens.primary"
+                  icon="info"
+                />
+              </router-link>
+              <KTooltip
+                :reference="`detailIcon${index}`"
+                :refs="$refs"
+                maxWidth="200px"
+              >
+                {{ $tr('details') }}
+              </KTooltip>
+            </div>
+
+            <ChannelStar
+              :channelId="channel.id"
+              :data-testid="`bookmark-button-${index}`"
+              :bookmark="channel.bookmark"
+            />
+            <KIconButton
+              size="small"
+              icon="optionsVertical"
+              appearance="flat-button"
+              :data-testid="`dropdown-button-${index}`"
+              @click.stop="openDropDown(channel, index)"
+            >
+              <template #menu>
+                <KDropdownMenu
+                  :hasIcons="true"
+                  :options="dropDownArr"
+                  @select="option => selectedItem(option, channel)"
+                />
+              </template>
+            </KIconButton>
+          </div>
+        </div>
+      </template>
+    </KCard>
+
+    <KModal
+      v-if="deleteDialog"
+      :title="canEdit ? $tr('deleteTitle') : $tr('removeTitle')"
+      :submitText="canEdit ? $tr('deleteChannel') : $tr('removeBtn')"
+      :cancelText="$tr('cancel')"
+      data-testid="delete-modal"
+      @submit="handleDelete"
+      @cancel="deleteDialog = false"
+    >
+      {{ canEdit ? $tr('deletePrompt') : $tr('removePrompt') }}
+    </KModal>
+    <ChannelTokenModal
+      v-if="selectedChannel && selectedChannel.published"
+      v-model="tokenDialog"
+      data-testid="copy-modal"
+      :channel="selectedChannel"
+      @copied="trackTokenCopy"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import { mapActions } from 'vuex';
+  import useKResponsiveWindow from 'kolibri-design-system/lib/composables/useKResponsiveWindow';
+  import { RouteNames } from '../../../constants';
+  import ChannelStar from '../ChannelStar';
+  import ChannelTokenModal from 'shared/views/channel/ChannelTokenModal';
+  import Languages from 'shared/leUtils/Languages';
+
+  export default {
+    name: 'StudioChannelCard',
+    components: {
+      ChannelStar,
+      ChannelTokenModal,
+    },
+    setup() {
+      const { windowIsSmall } = useKResponsiveWindow();
+      return {
+        windowIsSmall,
+      };
+    },
+    props: {
+      channel: {
+        type: Object,
+        required: true,
+      },
+      index: {
```

Comment:
Since a single card is now abstracted to a component, `index` can be removed. After you've re-organized test suites as suggested in the other comment, it won't be needed - e.g. instead of `:data-testid="dropdown-button-${index}"`, you can just `:data-testid="dropdown-button"`. And instead of `:ref="detailIcon${index}"`, you too can just `:ref="detailIcon"` - references should be scoped to this component.